diff --git a/node_modules/react-native-waterfall-flow/src/index.js b/node_modules/react-native-waterfall-flow/src/index.js
index fa6880a..a224440 100644
--- a/node_modules/react-native-waterfall-flow/src/index.js
+++ b/node_modules/react-native-waterfall-flow/src/index.js
@@ -1,10 +1,10 @@
 import React from 'react'
-import { View, FlatList, Platform } from 'react-native'
+import { View,  Platform } from 'react-native'
 import WaterfallItem from './WaterfallItem'
 const invariant = require('invariant')
+import { FlatList } from 'react-native-scroll-head-tab-view'
 
 export default class WaterfallFlow extends React.PureComponent {
-
   constructor(props) {
     super(props)
     // this._checkProps(props)
@@ -189,7 +189,6 @@ export default class WaterfallFlow extends React.PureComponent {
     this._itemHeights.length = data.length
 
     this._computePositions(true)
-
     return (
       <FlatList
         {...this.props}
