diff --git a/node_modules/react-native-pell-rich-editor/src/RichEditor.js b/node_modules/react-native-pell-rich-editor/src/RichEditor.js
index 74de536..b976847 100755
--- a/node_modules/react-native-pell-rich-editor/src/RichEditor.js
+++ b/node_modules/react-native-pell-rich-editor/src/RichEditor.js
@@ -218,7 +218,7 @@ export default class RichTextEditor extends Component {
         if (height !== this.state.height) {
             const maxHeight = Math.max(height, initialHeight);
             if (!this.unmount && useContainer && maxHeight >= initialHeight) {
-                this.setState({height: maxHeight});
+                this.setState({height: maxHeight + 40});
             }
             onHeightChange && onHeightChange(height);
         }
diff --git a/node_modules/react-native-pell-rich-editor/src/editor.js b/node_modules/react-native-pell-rich-editor/src/editor.js
index 4252bd9..d126b6b 100644
--- a/node_modules/react-native-pell-rich-editor/src/editor.js
+++ b/node_modules/react-native-pell-rich-editor/src/editor.js
@@ -559,6 +559,9 @@ function createHTML(options = {}) {
                     if (ele.checked) ele.setAttribute('checked', '');
                     else ele.removeAttribute('checked');
                 }
+                setTimeout(function () {
+                    Actions.UPDATE_OFFSET_Y();
+                });
             }
             addEventListener(content, 'touchcancel', handleSelecting);
             addEventListener(content, 'mouseup', handleSelecting);
